charts/super-agent-deployment/values.yaml

```diff
@@ -97,18 +97,65 @@ fedramp:
 verboseLog:
 
 config:
-  # -- Set if the configMap is going to be created by this chart or the user will provide its own.
-  create: true
-  # -- The name the configMap is going to have. If create is set to false, the name of an existing configMap that will be used to config the Super Agent.
-  name: ""
-  # -- The key in the configMap that has the configuration for the Super Agent.
-  key: ""
-  # -- Here you can set New Relic' Super Agent configuration.
+  # -- Configuration for the Super Agent.
+  # @default -- See `values.yaml`
+  superAgent:
+    # -- Set if the configMap is going to be created by this chart or the user will provide its own.
+    create: true
+    # -- The name the configMap is going to have. If create is set to false, the name of an existing configMap that will be used to config the Super Agent.
+    name: ""
+    # -- The key in the configMap that has the configuration for the Super Agent.
+    key: ""
+    # -- Here you can set New Relic' Super Agent configuration.
+    # @default -- See `values.yaml` for examples
+    content:
+    #  opamp:
+    #    headers:
+    #      some-key: some-value
+    #  agents:
+    #    infra-test:
+    #      agent_type: namespace/com.newrelic.infrastructure_agent:0.0.2
+    #    on-host-collector:
+    #      agent_type: namespace/io.opentelemetry.collector:0.0.1
+    #    kubernetes-collector:
+    #      agent_type: namespace/io.k8s.opentelemetry.collector:0.0.1
+
+  # -- Values that the fleet is going to have in the deployment.
   # @default -- See `values.yaml` for examples
-  content:
-  #  opamp:
-  #    headers:
-  #      some-key: some-value
-  #  agents:
-  #    agent_1:
-  #      agent_type: namespace/agent_type:0.0.1
+  fleet: {}
```

Comment:
why fleet and not subAgents if the one above is superAgent?

Reply:
Because there are too many agents: super agent, agent type, subagent,...

I got lost more than once, so I wanted to provide a better UX, so I called it fleet to open a conversation about it.

I have no opinion but I would love a new name for this.

Reply:
I agree, I don't particularly appreciate that we have many files and that it is not clear what they are used for
IMO the best would be to have it below `agent_type` as `agent_values` so that we do not need another tree and it is clear how it is used. But it is horrible that we do not need it when we create the SA file.
```yaml
config:
  superAgent:
    content:
       opamp:
           headers:
             some-key: some-value
        agents:
           infra-test:
             agent_type: namespace/com.newrelic.infrastructure_agent:0.0.2
             agent_values: blablabla (but this creates issues/confusion when creating the config file)
```


Then another "tree" is fine, but `fleet` is confusing in my opinion... In the end, if the one above is a `config.superAgent` having its `.agents` the one below could be the `config.subAgents` holding the config for all the agents specified above 🤔

Reply:
Fair enough.

Reply:
Change `fleet` to `subagent` on [#6d3db5f](https://github.com/newrelic/helm-charts/pull/1221/commits/6d3db5f906146aba70c299bef0cbc2cf62d43b12)